cmd/bosun/sched/host.go

```diff
@@ -0,0 +1,809 @@
+			if ts["host"] != host.Name {
+				continue
+			}
+			name, ok := ts["name"]
+			if !ok {
+				slog.Errorf("couldn't find name tag %s for host %s", host.Name, name)
+			}
+			tStatus, timestamp, err := s.Search.GetLast("hw.chassis.temps", ts.String(), false)
+			celsius, rTimestamp, err := s.Search.GetLast("hw.chassis.temps.reading", ts.String(), false)
+			status := "Bad"
+			if tStatus == 0 {
+				status = "Ok"
+			}
+			if err == nil && timestamp > 0 {
+				host.Hardware.Temps[name] = &Temp{
+					Celsius:            celsius,
+					Status:             status,
+					StatusLastUpdated:  timestamp,
+					CelsiusLastUpdated: rTimestamp,
+				}
+			}
+		}
+		for _, ts := range hwPowerSuppliesTags {
+			if ts["host"] != host.Name {
+				continue
+			}
+			id, ok := ts["id"]
+			if !ok {
+				return nil, fmt.Errorf("couldn't find power supply tag for host %s", host.Name)
+			}
+			idPlus, err := strconv.Atoi(id)
+			if err != nil {
+				slog.Errorf("couldn't conver it do integer for power supply id %s", id)
+			}
+			idPlus++
+			fstatus, timestamp, err := s.Search.GetLast("hw.ps", ts.String(), false)
+			status := "Bad"
+			if fstatus == 0 {
+				status = "Ok"
+			}
+			current, currentTimestamp, _ := s.Search.GetLast("hw.chassis.current.reading", opentsdb.TagSet{"host": host.Name, "id": fmt.Sprintf("PS%v", idPlus)}.String(), false)
+			volts, voltsTimestamp, _ := s.Search.GetLast("hw.chassis.volts.reading", opentsdb.TagSet{"host": host.Name, "name": fmt.Sprintf("PS%v_Voltage_%v", idPlus, idPlus)}.String(), false)
+			ps := &PowerSupply{}
+			if err == nil && timestamp > 0 {
+				ps.Status = status
+				ps.StatusLastUpdated = timestamp
+				ps.Amps = current
+				ps.AmpsLastUpdated = currentTimestamp
+				ps.Volts = volts
+				ps.VoltsLastUpdated = voltsTimestamp
+				host.Hardware.PowerSupplies[id] = ps
+			}
+			for _, m := range hostMetadata {
+				if m.Time.Before(time.Now().Add(-timeFilterAge)) {
+					continue
+				}
+				if !m.Tags.Equal(ts) {
+					continue
+				}
+				switch val := m.Value.(type) {
+				case string:
+					switch m.Name {
+					case "psMeta":
+						err = json.Unmarshal([]byte(val), &ps)
+						if err != nil {
+							slog.Errorf("error unmarshalling power supply meta for host %s, while generating host api: %s", host.Name, err)
+						} else {
+							host.Hardware.PowerSupplies[id] = ps
+						}
+					}
+				}
+			}
+		}
+		for _, ts := range hwBatteriesTags {
+			if ts["host"] != host.Name {
+				continue
+			}
+			id, ok := ts["id"]
+			if !ok {
+				slog.Errorf("couldn't find battery id tag %s for host %s", host.Name, id)
+				continue
+			}
+			fstatus, timestamp, err := s.Search.GetLast("hw.storage.battery", ts.String(), false)
+			status := "Bad"
+			if fstatus == 0 {
+				status = "Ok"
+			}
+			if err == nil && timestamp > 0 {
+				host.Hardware.Storage.Batteries[id] = &Battery{
+					Status:            status,
+					StatusLastUpdated: timestamp,
+				}
+			}
+		}
+		for _, ts := range hwBoardPowerTags {
```

Comment:
right after you fetch the data. Turn it into a `map[string][]Tagset`. only include tagsets with the tags you will be looking for later.